smart-open version 5.0.0 breaks serialization tests

### smart-open version 5.0.0 breaks serialization tests
The latest release of `smart-open` is causing some serialization tests to fail. We should investigate to determine if we can make updates that will work with both newer and older versions of smart-open or if we need to update the requirements permanently.


#### Stack Trace

<details>

```
___________________________________________________________________________________ test_s3_test_profile[pd_es] ___________________________________________________________________________________

es = Entityset: ecommerce
  Entities:
    régions [Rows: 2, Columns: 2]
    stores [Rows: 6, Columns: 3]
    products [Rows...régions.id
    sessions.customer_id -> customers.id
    log.session_id -> sessions.id
    log.product_id -> products.id
s3_client = s3.ServiceResource(), s3_bucket = s3.Bucket(name='test-bucket'), setup_test_profile = None

    def test_s3_test_profile(es, s3_client, s3_bucket, setup_test_profile):
        if not all(isinstance(entity.df, pd.DataFrame) for entity in es.entities):
            pytest.xfail('tmp file disappears after deserialize step, cannot check equality with Dask')
>       es.to_csv(TEST_S3_URL, encoding='utf-8', engine='python', profile_name='test')

featuretools/tests/entityset_tests/test_serialization.py:358: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
featuretools/entityset/entityset.py:201: in to_csv
    serialize.write_data_description(self, path, format='csv', index=False, sep=sep, encoding=encoding, engine=engine, compression=compression, profile_name=profile_name)
featuretools/entityset/serialize.py:150: in write_data_description
    use_smartopen_es(file_path, path, read=False, transport_params=transport_params)
featuretools/utils/s3_utils.py:15: in use_smartopen_es
    with open(path, 'wb', transport_params=transport_params) as fout:
env/lib/python3.7/site-packages/smart_open/smart_open_lib.py:221: in open
    binary = _open_binary_stream(uri, binary_mode, transport_params)
env/lib/python3.7/site-packages/smart_open/smart_open_lib.py:383: in _open_binary_stream
    fobj = submodule.open_uri(uri, mode, transport_params)
env/lib/python3.7/site-packages/smart_open/s3.py:196: in open_uri
    return open(parsed_uri['bucket_id'], parsed_uri['key_id'], mode, **kwargs)
env/lib/python3.7/site-packages/smart_open/s3.py:280: in open
    writebuffer=writebuffer,
env/lib/python3.7/site-packages/smart_open/s3.py:734: in __init__
    _initialize_boto3(self, client, client_kwargs, bucket, key)
env/lib/python3.7/site-packages/smart_open/s3.py:500: in _initialize_boto3
    client = boto3.client('s3', **init_kwargs)
env/lib/python3.7/site-packages/boto3/__init__.py:93: in client
    return _get_default_session().client(*args, **kwargs)
env/lib/python3.7/site-packages/boto3/session.py:263: in client
    aws_session_token=aws_session_token, config=config)
env/lib/python3.7/site-packages/botocore/session.py:797: in create_client
    region_name = self._resolve_region_name(region_name, config)
env/lib/python3.7/site-packages/botocore/session.py:851: in _resolve_region_name
    region_name = self.get_config_variable('region')
env/lib/python3.7/site-packages/botocore/session.py:241: in get_config_variable
    logical_name)
env/lib/python3.7/site-packages/botocore/configprovider.py:301: in get_config_variable
    return provider.provide()
env/lib/python3.7/site-packages/botocore/configprovider.py:398: in provide
    value = provider.provide()
env/lib/python3.7/site-packages/botocore/configprovider.py:459: in provide
    scoped_config = self._session.get_scoped_config()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <botocore.session.Session object at 0x1442a0fd0>

    def get_scoped_config(self):
        """
        Returns the config values from the config file scoped to the current
        profile.
    
        The configuration data is loaded **only** from the config file.
        It does not resolve variables based on different locations
        (e.g. first from the session instance, then from environment
        variables, then from the config file).  If you want this lookup
        behavior, use the ``get_config_variable`` method instead.
    
        Note that this configuration is specific to a single profile (the
        ``profile`` session variable).
    
        If the ``profile`` session variable is set and the profile does
        not exist in the config file, a ``ProfileNotFound`` exception
        will be raised.
    
        :raises: ConfigNotFound, ConfigParseError, ProfileNotFound
        :rtype: dict
    
        """
        profile_name = self.get_config_variable('profile')
        profile_map = self._build_profile_map()
        # If a profile is not explicitly set return the default
        # profile config or an empty config dict if we don't have
        # a default profile.
        if profile_name is None:
            return profile_map.get('default', {})
        elif profile_name not in profile_map:
            # Otherwise if they specified a profile, it has to
            # exist (even if it's the default profile) otherwise
            # we complain.
>           raise ProfileNotFound(profile=profile_name)
E           botocore.exceptions.ProfileNotFound: The config profile (test) could not be found

env/lib/python3.7/site-packages/botocore/session.py:340: ProfileNotFound
-------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------
WARNING:smart_open.utils:ignoring unsupported keyword arguments: ['session']
---------------------------------------------------------------------------------------- Captured log call ----------------------------------------------------------------------------------------
WARNING  smart_open.utils:utils.py:69 ignoring unsupported keyword arguments: ['session']
```

</details>

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -8,7 +8,7 @@ Release Notes
     * Changes
         * Minor updates to work with Koalas version 1.7.0 (:pr:`1351`)
         * Explicitly mention Python 3.8 support in setup.py classifiers (:pr:`1371`)
-        * Restrict smart-open version to <5.0.0 (:pr:`1372`)
+        * Fix issue with smart-open version 5.0.0 (:pr:`1372`)
     * Documentation Changes
     * Testing Changes
          * Make release notes updated check separate from unit tests (:pr:`1347`)

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -8,7 +8,7 @@ Release Notes
     * Changes
         * Minor updates to work with Koalas version 1.7.0 (:pr:`1351`)
         * Explicitly mention Python 3.8 support in setup.py classifiers (:pr:`1371`)
-        * Fix issue with smart-open version 5.0.0 (:pr:`1372`)
+        * Fix issue with smart-open version 5.0.0 (:pr:`1372`, :pr:`1376`)
     * Documentation Changes
     * Testing Changes
          * Make release notes updated check separate from unit tests (:pr:`1347`)

--- a/featuretools/utils/s3_utils.py
+++ b/featuretools/utils/s3_utils.py
@@ -61,7 +61,7 @@ BOTOCORE_ERR_MSG = (
 SMART_OPEN_ERR_MSG = (
     "The smart_open library is required to read and write from URLs and S3.\n"
     "Install via pip:\n"
-    "    pip install smart-open\n"
+    "    pip install 'smart-open>=5.0.0'\n"
     "Install via conda:\n"
-    "    conda install smart_open"
+    "    conda install 'smart-open>=5.0.0'"
 )

--- a/featuretools/utils/s3_utils.py
+++ b/featuretools/utils/s3_utils.py
@@ -33,11 +33,12 @@ def get_transport_params(profile_name):
     Config = import_or_raise("botocore.config", BOTOCORE_ERR_MSG).Config
 
     if isinstance(profile_name, str):
-        transport_params = {'session': boto3.Session(profile_name=profile_name)}
+        session = boto3.Session(profile_name=profile_name)
+        transport_params = {'client': session.client('s3')}
     elif profile_name is False or boto3.Session().get_credentials() is None:
-        transport_params = {
-            'resource_kwargs': {'config': Config(signature_version=UNSIGNED)}
-        }
+        session = boto3.Session()
+        client = session.client('s3', config=Config(signature_version=UNSIGNED))
+        transport_params = {'client': client}
     else:
         transport_params = None
     return transport_params

--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -6,7 +6,7 @@ pytest-cov==2.6.1
 fastparquet>=0.1.6
 graphviz>=0.8.4
 moto[all]>=1.3.15
-smart-open>=1.8.4,<5.0.0
+smart-open>=5.0.0
 boto3>=1.10.45
 composeml>=0.2.0
 urllib3>=1.25.10,<1.26

